### ISSUE TITLE
Incorrect report of: call to unsafe function

### ISSUE THREAD

**Issue by dream-dasher** [rust-lang/rust-analyzer#18850]
The attached code incorrectly (?) reports a call to an unsafe function.
This is only shown in-IDE (Zed or Helix).  It is not shown with `cargo check` or `cargo clippy`.
The code in question is just a call to produce a new array, with populated values.
(There is no visible issue at runtime.)

I was able to get the code down to what looks like any vec of arrays.

In-IDE error: 
```
rust-analyzer: call to unsafe function is unsafe and requires an unsafe function or block
```
Minimal-Example-Helix:
<img width="800" alt="Image" src="https://github.com/user-attachments/assets/baee3847-e6bd-47a8-942a-dc2c87222013" />
Helix:
<img width="400" alt="Image" src="https://github.com/user-attachments/assets/0b061bbd-8385-4373-845e-d47d93e497b2" />
Zed:
<img width="400" alt="Image" src="https://github.com/user-attachments/assets/2c8a8465-94b2-4c32-bdb7-6fcc90ec5b3c" />

with Zed or Helix:

**rust-analyzer** 1.86.0-nightly (b3b368a1 2025-01-05)
**rustc** 1.86.0-nightly (b3b368a18 2025-01-05)
**Zed** 0.167.1 – /Applications/Zed.app
**helix** 25.1 (dabfb6ce)

```rust
fn main() {
        let an_array: [bool; 3] = [false; 3];
        let vec_of_arrays = vec![an_array];
        print!("{:?}", vec_of_arrays);
}
```

Here's the minimal example as a cargo-script: [dbg_incorrect-unsafe-error-minimal.rs](https://github.com/dream-dasher/cargo-scripts/blob/main/dbg_incorrect_unsafe_error-minimal.rs)
And (probably not needed) here's the orignating non-minimal example as a cargo-script: [dbg_incorrect-unsafe-error.rs](https://github.com/dream-dasher/cargo-scripts/blob/main/dbg_incorrect_unsafe_error.rs)
(I realize that rust-analyzer doesn't yet work with cargo-script at the time of posting. But hopefully soon! Figured it would still be best practice. :)

**Comment by Veykril** [comment]
Duplicate of https://github.com/rust-lang/rust-analyzer/issues/18842
